### PR TITLE
Private event Project

### PR DESCRIPTION
**Pull request by Emmyn5600**
### This pull request contains the following features:

* ### Updated Readme file
* ### Created form for signup and sign in before creating or view an event
* ### User should create an event after signing or signing up 
* ### User should click attend the event button to attend the event
* ### user should cancel or ignore to attend the event 
* ### user should see the events they have created as well as events they are going to attend.

### CODE REVIEW

File: db/migrate/20210503194732_create_events.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreateEvents < ActiveRecord::Migration[6.1]
+  def change
+    create_table :events do |t|
+      t.string :title
+      t.text :description
+      t.string :location
+      t.datetime :date
+      t.integer :creator_id
```
Comment (descholar-ceo) [inline db/migrate/20210503194732_create_events.rb:8]: - [x] [required] Events table in migrations has creator_id as the foreign key ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from Odin)

File: app/views/user/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<% if @user.attended_events.any? %>
+	<h2>Upcoming Events</h2> 
+	<ul class="media-list">
+		<%= render @upcoming_events %>
+	</ul>
+<% else %>
+	<h2>No Upcoming Events</h2>
+<% end %>
```
Comment (descholar-ceo) [inline app/views/user/show.html.erb]: - [x] [Required] Kindly remove this logic from the views, you can always use helper methods for all of the view methods

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,16 +1,67 @@
...
+                <p>
+                  <%= link_to "Logout", destroy_user_session_path, method: :delete, class:"button is-primary" %>
+                </p>
+              <% else %>
+              <p class="control">
+                <%= link_to 'Sign In', new_user_session_path, class: "button is-primary" %>
+              </p>
+              <p class="control">
+                <%= link_to 'Sign Up', new_user_registration_path, class: "button is-primary" %>
+              </p>
```
Comment (descholar-ceo) [inline app/views/layouts/application.html.erb]: - [x] [Required] Kindly remove this logic from the views, you can always use helper methods for all of the view methods

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @Emmyn5600 and @VanessaAoki  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implement all of this project requirements so carefully, keep it up, and your project is really almost done.
:+1: I liked your User interface, it's really awesome.
:+1: Your readme file is also professional and it shows your potential of attention to details, keep up the good job guys :muscle: 
#### :warning: However we need to make few changes to mark this project approved. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [x] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming (covered by automated tests) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from Odin)
=> Currently you don't have a user's showpage link, that route is even missing from the routes of this project, kindly fix this issue. 
- [x] Models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from Pathwright)
- [x] Authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from Pathwright)
- [x] Event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from Pathwright)
- [x] There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers. View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb). There shoud be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it.
=> To help you understand what to change here, is that you need to move all of if conditions statement, data manipulations, and loops from all of the `.html.erb` files and move those logic to `app/helpers/*_helper.rb` files 

# [Optional changes] :yellow_circle: 
- [x] It would be very nice if you hide the new event button if the user is not signed in
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Review by ElijahTobs — APPROVED** [review]
# Status: Approved! :tada: 

## Congratulations @Emmyn5600 and @VanessaAoki  

### Good job implementing the changes requested by the previous TSE.

## You can now merge and submit.

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Pull request merged**